### ISSUE TITLE
Detect when tracking is lost

### ISSUE THREAD

**Issue by TomiAijala** [NVlabs/FoundationPose#136]
Hello. Is there a way to detect when tracking of an object is lost and the pose needs to be re-initialized? Is there a confidence score or something similar available for the produced pose during tracking to help with detecting when tracking is no longer reliable?

**Comment by wenbowen123** [comment]
Hi, one thing to try is to send the pose to score network and see the score. Then set some threshold accordingly.

**Comment by assia855** [comment]
Hi @wenbowen123  how can I do that please I'm struggling with the same issue.

**Comment by Marco2929** [comment]
> Hi, one thing to try is to send the pose to score network and see the score. Then set some threshold accordingly.

The problem I have with this approach is that the score varies too much. Even if the pose is still correct, the scorer returns a completely different number due to the slightly different orientation of the object. I tried to use the initial score and make use of the deviation.

**Comment by PannagaS** [comment]
@Marco2929 did you figure out how to re-initialize the pose based on scores corresponding to occlusion or when there is loss of tracking?

**Comment by Marco2929** [comment]
@PannagaS Yes, I used two different thresholds. The first detects whether the object is occluded or missing, as the score rises quickly in such cases. The second ensures that reinitialization worked correctly and that the object is back in the scene, where the score tends to return to its original value. One problem is that repeated reinitialization lowers the frame rate.

**Comment by PannagaS** [comment]
@Marco2929  Yeah, even I observed that repeated reinitialization lowers the frame rate. Can you share with me that block of code where you included thresholds? Is it in `def track_one()` in `estimator.py` ? I desperately need help with this.
